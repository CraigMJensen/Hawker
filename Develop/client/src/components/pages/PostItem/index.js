import React, { useState } from 'react';
import { useMutation } from '@apollo/client';
import { ADD_ITEM } from '../../../utils/mutations';
import Auth from '../../../utils/auth';

function PostItem() {
  const [postForm, setPostForm] = useState('');
  const [addPost] = useMutation(ADD_ITEM);

  const categories = [
    { name: 'Outdoor', id: '6256096b98fc0602b12202fb' },
    { name: 'Transportation', id: '6256096b98fc0602b12202fc' },
    { name: 'Tech', id: '6256096b98fc0602b12202fd' },
    { name: 'Sports', id: '6256096b98fc0602b12202fe' }
  ];

  const handleFormSubmit = async (event) => {
    event.preventDefault();
    const mutationResponse = await addPost({
      variables: {
        title: postForm.title,
        description: postForm.description,
        image: postForm.image,
        price: postForm.price,
        quanity: postForm.quanity,
        category: postForm.catergory
      }
    });
    const token = mutationResponse.data.addPost.token;
    Auth.login(token);
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setPostForm({
      ...postForm,
      [name]: value
    });
  };

  return (
    <div>
      <form onSubmit={handleFormSubmit}>
        <label for="itemImage">Insert Image </label>
        <input type="file" name="itemImage" onChange={handleChange}></input>

        <label for="itemTitle">Item Title</label>
        <input type="text" name="itemTitle" placeholder="Title" onChange={handleChange}></input>

        <label for="price">Price</label>
        <input type="text" name="price" placeholder="Price of Item" onChange={handleChange}></input>

        <label for="description">Description</label>
        <input
          type="text"
          name="description"
          placeholder="Description"
          onChange={handleChange}
        ></input>

        <label for="description">Quantity</label>
        <input type="Number" name="Quantity" placeholder="Quantity" onChange={handleChange}></input>
        
       <select> {categories.map((category) => { return <option>{category.name + ' - ' +  category.id}</option>})}</select>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
}

export default PostItem;
