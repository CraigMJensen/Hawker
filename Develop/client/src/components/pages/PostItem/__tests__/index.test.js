// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PostItem from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('PostItem component', () => {
  // verify the component is rendering properly
  it('PostItem component renders', () => {
    render(<PostItem />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<PostItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});