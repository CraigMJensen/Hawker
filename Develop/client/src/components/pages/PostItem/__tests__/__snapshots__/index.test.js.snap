// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Post Item
    </h2>
  </div>
</DocumentFragment>
`;
